src/libraries/IterableMap.sol

```diff
@@ -0,0 +1,104 @@
+pragma solidity ^0.4.13;
+
+import 'ROOT/Controller.sol';
+
+contract IterableMap {
+
+    struct Item {
+        bool hasItem;
+        uint256 value;
+        uint256 offset;
+    }
+
+    address owner;
+    bool initialized;
+    uint256[] itemsArray;
+    Item[] itemsMap;
+    uint numberOfItems;
+
+    function any() {
```

Comment:
It does not, it has modifiers though which can be used to achieve similar effects (and modifiers can even take function parameters too and use them if needed)